src/linter/conf.go

```diff
@@ -68,6 +75,82 @@ type Config struct {
 	StrictMixed bool
 }
 
+type RuleNode struct {
+	Children map[string]*RuleNode // Child nodes (subdirectories and files)
+	Enable   map[string]bool      // Rules enabled at this level
+	Disable  map[string]bool      // Disabled rules at this level
+}
+
+func NewRuleNode() *RuleNode {
+	return &RuleNode{
+		Children: make(map[string]*RuleNode),
+		Enable:   make(map[string]bool),
+		Disable:  make(map[string]bool),
+	}
+}
+
+type PathRuleSet struct {
+	Enable  map[string]bool // Rules that are enabled for this path
+	Disable map[string]bool // Rules that are disabled for this path
+}
+
+func BuildRuleTree(pathRules map[string]*PathRuleSet) *RuleNode {
+	root := NewRuleNode()
+
+	for path, ruleSet := range pathRules {
+		normalizedPath := filepath.ToSlash(filepath.Clean(path))
+		parts := strings.Split(normalizedPath, "/")
+		currentNode := root
+
+		for _, part := range parts {
+			if part == "" {
+				continue
+			}
+			if _, exists := currentNode.Children[part]; !exists {
+				currentNode.Children[part] = NewRuleNode()
+			}
+			currentNode = currentNode.Children[part]
+		}
+
+		for rule := range ruleSet.Enable {
+			currentNode.Enable[rule] = true
+		}
+		for rule := range ruleSet.Disable {
+			currentNode.Disable[rule] = true
+		}
+	}
+
+	return root
+}
+
+func IsRuleEnabled(root *RuleNode, filePath string, checkRule string) bool {
+	normalizedPath := filepath.ToSlash(filepath.Clean(filePath))
+	parts := strings.Split(normalizedPath, "/")
+	currentNode := root
+
+	// Starting with global state. We have guarantee while parsing config that rule is `on` and exist
+	ruleState := true
+
+	for _, part := range parts {
+		if part == "" {
```

Comment:
is this exactly possible?

Reply:
I think yes:
1. filePath := "/A/B/C" => ["", "A", "B", "C"]
2. filePath := "A//B/C" and "A/B/C/". We used filepath.Clean, but if something will be wrong...idkn
I think that enabling this check ensures that our code will correctly handle any paths, regardless of their origin or format